core/tests/protractor_desktop/explorationImprovementsTab.js

```diff
@@ -0,0 +1,153 @@
+    creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+    explorationPlayerPage =
+      new ExplorationPlayerPage.ExplorationPlayerPage();
+
+    users.createUser(
+      'learner@user.com', 'learnerUser');
+    users.createAndLoginAdminUser(
+      'creator@user.com', 'creatorUser');
+
+    // Creator creates and publishes an exploration.
+    workflow.createExplorationAsAdmin();
+    explorationEditorMainTab.exitTutorial();
+
+    explorationEditorPage.navigateToSettingsTab();
+    explorationEditorSettingsTab.setTitle(EXPLORATION_TITLE);
+    explorationEditorSettingsTab.setCategory(EXPLORATION_CATEGORY);
+    explorationEditorSettingsTab.setObjective(EXPLORATION_OBJECTIVE);
+    explorationEditorSettingsTab.setLanguage(EXPLORATION_LANGUAGE);
+
+    explorationEditorPage.navigateToMainTab();
+    explorationEditorMainTab.setStateName('One');
+    explorationEditorMainTab.setContent(
+      forms.toRichText('Please write 1 in words.'));
+    explorationEditorMainTab.setInteraction('TextInput');
+    explorationEditorMainTab.addResponse(
+      'TextInput', forms.toRichText('Good job'), 'End', true, 'Equals',
+      'One');
+    explorationEditorMainTab.getResponseEditor('default').setFeedback(
+      forms.toRichText('Try again'));
+    explorationEditorMainTab.setSolicitAnswerDetailsFeature();
+    explorationEditorMainTab.moveToState('End');
+    explorationEditorMainTab.setInteraction('EndExploration');
+    explorationEditorPage.saveChanges();
+    workflow.publishExploration();
+
+    adminPage.editConfigProperty(
+      'Always ask learners for answer details while playing exploration.',
+      'Boolean',
+      function(elem) {
+        elem.setValue(true);
+      });
+
+    adminPage.editConfigProperty(
+      'Exposes the Improvements Tab for creators in the exploration editor',
+      'Boolean',
+      function(elem) {
+        elem.setValue(true);
+      });
+  });
+
+  it('checks solicit answer details feature', function() {
+    users.login('learner@user.com');
+    libraryPage.get();
+    libraryPage.findExploration(EXPLORATION_TITLE);
+    libraryPage.playExploration(EXPLORATION_TITLE);
+    explorationPlayerPage.submitAnswer('TextInput', 'One');
+    explorationPlayerPage.submitAnswerDetails('I liked this choice of answer');
+    explorationPlayerPage.expectExplorationToNotBeOver();
+
+    oppiaLogo.click();
+    general.acceptAlert();
+    users.logout();
+    users.login('creator@user.com');
+    creatorDashboardPage.get();
+    creatorDashboardPage.navigateToExplorationEditor();
+    explorationEditorPage.navigateToImprovementsTab();
+    explorationEditorImprovementsTab.checkAnswerDetailsCard('One', '1');
+    explorationEditorImprovementsTab.navigateReviewAnswerDetails();
+    explorationEditorImprovementsTab.verifyAnswerDetails(
+      'I liked this choi...');
+    users.logout();
+    users.createAndLoginAdminUser('testadm@collections.com', 'testadm');
+    adminPage.get();
```

Comment:
You should not turn these on in a beforeAll() and turn them off within a test.

They should be turned on in a beforeAll() and turned off in an afterAll(). Or turned on in a test and turned off at the end of the same test. But don't mix and match.

Reply:
Done